com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ServiceRemovalCalculator.java

```diff
@@ -0,0 +1,72 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceGetter;
+import com.sap.cloud.lm.sl.cf.core.model.ServiceInstanceWithServiceKeys;
+import com.sap.cloud.lm.sl.cf.process.Messages;
+import com.sap.cloud.lm.sl.cf.process.analytics.model.ServiceAction;
+import com.sap.cloud.lm.sl.cf.process.steps.ExecutionWrapper;
+import com.sap.cloud.lm.sl.cf.process.steps.StepsUtil;
+
+@Named
+public class ServiceRemovalCalculator {
+
+    private ServiceGetter serviceGetter;
+
+    @Inject
+    public ServiceRemovalCalculator(ServiceGetter serviceGetter) {
+        this.serviceGetter = serviceGetter;
+    }
+
+    public List<ServiceInstanceWithServiceKeys> calculateServicesToDelete(ExecutionWrapper execution,
+                                                                          List<CloudServiceInstance> serviceInstances) {
+        List<ServiceInstanceWithServiceKeys> serviceInstancesWithServiceKeys = getServiceInstancesWithServiceKeys(execution.getControllerClient(),
+                                                                                                                  serviceInstances);
+        if (StepsUtil.getServiceActionsToExecute(execution.getContext())
```

Comment:
can you doublecheck if this is correct? The service action should be for 1 service, but the signature of the method is a list of services.